### PR TITLE
Adding border radius full to `Box` component

### LINKED ISSUE #16116: Add 9999px border radius option to Box component
### Description
There is a need to allow the Box component to have a full rounded border radius.

### Technical Details
TBC

### Acceptance criteria 
TBC

### PR DESCRIPTION
**Pull request by georgewrmarshall**
## Explanation

Currently, there is no border radius prop option for a "full" pill like border radius.

This is a problem because currently engineers have to add extra css to achieve this and there are many instances where this would be useful

In order to solve this problem, this pull request adds a `full`(`9999px`) border radius option. It also separates the options from the main `SIZE` const in the design-system constants. It also updates tests, docs and stories.

## More Information

* Fixes #16116 
Some naming conventions from other libraries

### After

Screen Shot 2022-10-06 at 6 11 50 PM
Screen Shot 2022-10-06 at 6 12 43 PM

## Manual Testing Steps

- Go to the latest build of storybook on this PR
- Search `Box` component in search bar
- Check storybook controls for borderRadius work
- Check documentation for borderRadius in the Docs tab

## Pre-Merge Checklist

- [x] PR template is filled out
- [x] **IF** this PR fixes a bug, a test that _would have_ caught the bug has been added
- [x] PR is linked to the appropriate GitHub issue
- [x] PR has been added to the appropriate release Milestone

### + If there are functional changes:

- [x] Manual testing complete & passed
- [x] "Extension QA Board" label has been applied

### I will update all the current uses of the SIZE const for borderRadius and deprecate SIZES.NONE in another PR

Screen Shot 2022-10-06 at 4 58 35 PM
Screen Shot 2022-10-06 at 5 00 42 PM

### CODE REVIEW

File: ui/components/ui/box/README.mdx [hunk 1]
```diff
@@ -14,36 +14,36 @@ Box is a utility component that can be used for layout or as a base for other UI
...
+| borderWidth     | 0, 1, 2, 4, 6, 8, 9, 10, 12 or array of numbers [1, 2] for responsive props                                                                                                                                                           | -                  |
+| borderRadius    | BORDER_RADIUS values or array of BORDER_RADIUS values for responsive props from [../ui/helpers/constants/design-system.js](https://github.com/MetaMask/metamask-extension/blob/develop/ui/helpers/constants/design-system.js)         | -                  |
+| borderStyle     | BORDER_STYLE values or array of BORDER_STYLE values for responsive props from [../ui/helpers/constants/design-system.js](https://github.com/MetaMask/metamask-extension/blob/develop/ui/helpers/constants/design-system.js)           | -                  |
+| as              | The polymorphic `as` prop allows you to change the root HTML element of the Box component. Defaults to 'div'                                                                                                                          | 'div'              |
```
Comment (georgewrmarshall) [inline ui/components/ui/box/README.mdx:47]: Realized that links to blobbing although probably quite helpful. Become rapidly out of date as we edit or update the design-system.js file so updating to link to the file in general

Screen Shot 2022-10-06 at 6 12 17 PM

File: ui/components/ui/box/box.test.js [hunk 2]
```diff
@@ -9,8 +9,8 @@ import {
   ALIGN_ITEMS,
   JUSTIFY_CONTENT,
   TEXT_ALIGN,
-  SIZES,
```
Comment (georgewrmarshall) [inline ui/components/ui/box/box.test.js:12]: I'm just as amazed as you are the we only use `SIZES` for `bordeRadius` in the Box component 😮

File: ui/components/ui/box/box.scss [hunk 3]
```diff
@@ -134,23 +134,27 @@ $attributesToApplyExtraProperties: margin;
...
+    border-radius: 8px;
   }
 
   &--rounded-xl {
-    border-radius: 0.75rem;
+    border-radius: 12px;
+  }
+
+  &--rounded-pill {
+    border-radius: 9999px;
```
Comment (georgewrmarshall) [inline ui/components/ui/box/box.scss:157]: Updating to pixels here for less cognitive load when making the calculation to rems. There isn't any benefit in using rems outside of font sizes.

**Review by brad-decker — APPROVED** [review]
Awesome.

**Review by NidhiKJha — APPROVED** [review]

**Pull request merged**